pkg/result/processors/nolint.go

```diff
@@ -161,14 +161,16 @@ func (p *Nolint) shouldPassIssue(i *result.Issue) (bool, error) {
 		return false, err
 	}
 
+	matchesAtLeastOneRange := false
 	for _, ir := range fd.ignoredRanges {
 		if ir.doesMatch(i) {
+			// for nolintlint, we need to consider every range
 			ir.matchedIssueFromLinter[i.FromLinter] = true
-			return false, nil
+			matchesAtLeastOneRange = true
```

Comment:
I don't have test support for this.  It seemed like it would be pretty complicated to construct but I'm open to suggestions.  

This was needed because when I converted golangci lint to use machine-readable style directives (no leading space), the `interfacer` linter created a problem at https://github.com/golangci/golangci-lint/blob/master/pkg/golinters/scopelint.go#L166 (although this line itself did not change).

Running with `env GL_DEBUG=nolint ./golangci-lint run -E interfacer --disable-all`, you can see the nolint range getting expanded to include the entire function:

```
DEBU [nolint] file pkg/golinters/scopelint.go: inline nolint ranges are [{linters:[gocyclo gocritic] matchedIssueFromLinter:map[] Range:{From:73 To:76} col:1} {linters:[interfacer] matchedIssueFromLinter:map[] Range:{From:163 To:166} col:1}]
DEBU [nolint] found range is {[gocyclo gocritic] map[] {73 76} 1} for node &ast.FuncDecl{Doc:(*ast.CommentGroup)(0xc002b978c0), Recv:(*ast.FieldList)(0xc002b87ef0), Name:(*ast.Ident)(0xc002b97960), Type:(*ast.FuncType)(0xc0007776e0), Body:(*ast.BlockStmt)(0xc001e18b10)} [77;161], expanded range is {[gocyclo gocritic] map[] {73 161} 1}
DEBU [nolint] found range is {[gocyclo gocritic] map[] {73 76} 1} for node &ast.FuncType{Func:1966, Params:(*ast.FieldList)(0xc002b87f20), Results:(*ast.FieldList)(0xc002b87f50)} [77;77], expanded range is {[gocyclo gocritic] map[] {73 77} 1}
DEBU [nolint] found range is {[interfacer] map[] {163 166} 1} for node &ast.FuncDecl{Doc:(*ast.CommentGroup)(0xc0007776c0), Recv:(*ast.FieldList)(0xc001e18ba0), Name:(*ast.Ident)(0xc000777760), Type:(*ast.FuncType)(0xc000777ae0), Body:(*ast.BlockStmt)(0xc001e18c60)} [167;170], expanded range is {[interfacer] map[] {163 170} 1}
DEBU [nolint] found range is {[interfacer] map[] {163 166} 1} for node &ast.FuncType{Func:4213, Params:(*ast.FieldList)(0xc001e18c00), Results:(*ast.FieldList)(nil)} [167;167], expanded range is {[interfacer] map[] {163 167} 1}
DEBU [nolint] file pkg/golinters/scopelint.go: built nolint ranges are [{linters:[gocyclo gocritic] matchedIssueFromLinter:map[] Range:{From:73 To:76} col:1} {linters:[interfacer] matchedIssueFromLinter:map[] Range:{From:163 To:166} col:1} {linters:[gocyclo gocritic] matchedIssueFromLinter:map[] Range:{From:73 To:161} col:1} {linters:[gocyclo gocritic] matchedIssueFromLinter:map[] Range:{From:73 To:77} col:1} {linters:[interfacer] matchedIssueFromLinter:map[] Range:{From:163 To:170} col:1} {linters:[interfacer] matchedIssueFromLinter:map[] Range:{From:163 To:167} col:1}]
```

I'm still not completely clear on what changed but I think it makes sense that the expanded ranges would be treated as a group for the purpose of determining unused nolint directives.